lms/djangoapps/instructor/views/api.py

```diff
@@ -726,6 +728,343 @@ def extract_user_info(user):
+    else:
+        query_type = QUERY_TYPE.PROBLEM
+        if query_filtering == PROBLEM_FILTERS.OPENED.lower():
+            query_filtering = PROBLEM_FILTERS.OPENED
+        elif query_filtering == PROBLEM_FILTERS.NOT_OPENED.lower():
+            query_filtering = PROBLEM_FILTERS.NOT_OPENED
+        elif query_filtering == PROBLEM_FILTERS.COMPLETED.lower():
+            query_filtering = PROBLEM_FILTERS.COMPLETED
+        elif query_filtering == PROBLEM_FILTERS.NOT_COMPLETED.lower():
+            query_filtering = PROBLEM_FILTERS.NOT_COMPLETED
+        elif query_filtering == "score":
+            query_filtering = PROBLEM_FILTERS.SCORE
+        elif query_filtering == "number of peer responses graded":
+            query_filtering = PROBLEM_FILTERS.NUMBER_PEER_GRADED
+    return Query(query_type, query_incl, query_id, query_filtering, entity_name)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def delete_bulk_temp_query(request, course_id):  # pylint: disable=unused-argument
+    """
+    Deletes a temporary query that the user has entered along with the corresponding students
+    """
+    existing = request.GET.get('existing')
+    if existing is not None and existing != "":
+        existing_queries = existing.split(',')
+    else:
+        existing_queries = []
+    cleaned_queries = [query.strip() for query in existing_queries]
+    if len(cleaned_queries) > 0:
+        data_access.delete_bulk_temporary_query(cleaned_queries)
+    response_payload = {
+        'success': True
+    }
+    return JsonResponse(response_payload)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def delete_temp_query(_request, course_id, query_to_delete):  # pylint: disable=unused-argument
+    """
+    Deletes a temporary query that the user has entered along with the corresponding students
+    """
+    data_access.delete_temporary_query(query_to_delete)
+    response_payload = {
+        'success': True
+    }
+    return JsonResponse(response_payload)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def delete_saved_query(_request, course_id, query_to_delete):  # pylint: disable=unused-argument
+    """
+    Deletes a grouped query that the user has saved along with the corresponding subqueries
+    """
+    data_access.delete_saved_query(query_to_delete)
+    response_payload = {
+        'success': True
+    }
+    return JsonResponse(response_payload)
+
+
+@ensure_csrf_cookie
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_level('instructor')
+def save_query(request, course_id):
+    """
+    Saves a group of queries and assigns them the same group ID
+    """
+    existing = request.GET.get('existing')
+    if existing is not None and existing != "":
```

Comment:
This can be simplified to `if existing:` since both `None` and `""` evaluate to `Falsey`.